src/main/scala/db/migration/V2__convert_subjects_to_object.scala

```diff
@@ -0,0 +1,66 @@
+package db.migration
+import java.sql.Connection
+
+import io.circe.generic.semiauto._
+import io.circe.generic.auto._
+import io.circe.parser._
+import io.circe.syntax._
+import io.circe.{Decoder, Encoder}
+import no.ndla.frontpageapi.model.domain.{FrontPageData, SubjectCollection, SubjectFilters}
+import no.ndla.frontpageapi.repository._
+import org.flywaydb.core.api.migration.jdbc.JdbcMigration
+import org.postgresql.util.PGobject
+import scalikejdbc._
+
+/**
+  * Part of GDL frontpage-api.
+  * Copyright (C) 2018 GDL
+  *
+  * See LICENSE
+  */
+class V2__convert_subjects_to_object extends JdbcMigration {
+
+  implicit val decoder: Decoder[V2_DBFrontPageData] = deriveDecoder
+  implicit val encoder: Encoder[V2_DBFrontPageData] = deriveEncoder
+
+  override def migrate(connection: Connection): Unit = {
+    val db = DB(connection)
+    db.autoClose(false)
+    db.withinTx { implicit session =>
+      frontPageData.map(convertSubjects).foreach(update)
+    }
+  }
+
+  def frontPageData(implicit session: DBSession): Option[V2_DBFrontPage] = {
+    sql"select id, document from mainfrontpage"
+      .map(rs => V2_DBFrontPage(rs.long("id"), rs.string("document")))
+      .single
+      .apply
+  }
+
+  def convertSubjects(frontPage: V2_DBFrontPage): FrontPageData = {
```

Comment:
Tror det er smart å lage en kopi av `FrontPageData`, `V2_FrontPageData`, som vi bruker i migreringsscriptet.
Sett at scriptet skal kjøres på et sted som ligger mange versjoner bak. Da er det ikke gitt at `no.ndla.forntpageapi.model.domain.FrontPageData` er lik den versjonen som er der akkurat nå. Alle case classes som brukes i migreringsscript bør gjenspeile, men være helt isolerte fra omverdnen :)

Fungerer `convertSubjects` på et dokument som allerede er konvertert til det nye formatet?

Reply:
Dette skjer allerede. Koden som parser json til objekt feiler dersom json ikkje er på forventa format. Derfor vil ikkje migreringa kjøre uten av json er korrekt.

Reply:
Skjønte kva du meinte. La til caseklasser for at migreringsscriptet skriver til det formatet vi ønsker i denne versjonen av apiet.